core/core-impl/src/main/java/ai/timefold/solver/core/config/util/ConfigUtils.java

```diff
@@ -487,10 +487,28 @@ private static <C> Member getSingleMember(Class<C> clazz, Class<? extends Annota
         if (memberList.isEmpty()) {
             return null;
         }
-        if (memberList.size() > 1) {
-            throw new IllegalArgumentException("The class (" + clazz
-                    + ") has " + memberList.size() + " members (" + memberList + ") with a "
-                    + annotationClass.getSimpleName() + " annotation.");
+        int size = memberList.size();
+        if (clazz.isRecord()) {
+            /*
+             * A record has a field and a getter for each record component.
+             * When the component is annotated with @PlanningId,
+             * the annotation ends up both on the field and on the getter.
+             */
+            if (size == 2) { // The getter is used to retrieve the value of the record component.
+                return memberList.get(1);
```

Comment:
Is it guaranteed that the second member in the list is the getter and not the field?

Reply:
I don't think there is such guarantee. In fact, I'd expect that this is totally up to the discretion of every JDK implementation. Maybe I can figure out which is which from the type...